cmd/contour/serve.go

```diff
@@ -478,23 +482,25 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		log.Warn("DEPRECATED: The flag '--experimental-service-apis' is deprecated and should not be used. Please configure the gateway.name & gateway.namespace in the configuration file to specify which Gateway Contour will be watching.")
 	}
 
-	foundGatewayAPI := false
-	for _, r := range k8s.GatewayAPIResources() {
-		if !clients.ResourcesExist(r) {
-			log.WithField("resource", r).Warn("resource type not present on API server")
-			continue
-		}
-		foundGatewayAPI = true
-
-		if err := informOnResource(clients, r, &dynamicHandler); err != nil {
-			log.WithError(err).WithField("resource", r).Fatal("failed to create informer")
-		}
-	}
+	// Only inform on GatewayAPI resources if Gateway API is found.
+	if ctx.Config.GatewayConfig != nil {
+		if clients.ResourcesExist(k8s.GatewayAPIResources()...) {
+			for _, r := range k8s.GatewayAPIResources() {
+				if !clients.ResourcesExist(r) {
+					log.WithField("resource", r).Warn("resource type not present on API server")
+					continue
+				}
```

Comment:
this is now redundant with L487 that checks that all the resources exist. I'd probably remove this one.

Reply:
👍